content/topics/Analyze/machine-learning/supervised/K-fold_crossvalidation.md

```diff
@@ -0,0 +1,103 @@
+draft: false
+weight: 20.1
+author: "Kheiry Sohooli"
+authorlink: "https://tilburgsciencehub.com/contributors/kheirysohooli/" 
+---
+
+## Overview
+In the realm of Machine learning, model evaluation is an crucial for assessing their reliability. It helps in detecting overfitting and underfitting after training algorithms on datasets. Cross-validation stands out as the best approach to pinpoint overfitting.
+In this short article, you'll become familiar with the primary causes of overfitting and briefly introduce solutions to address it. Additionally, we'll explore K-fold cross-validation, the main method for detecting overfitting.
+
+## Overfitting 
+Overfitting occurs when a model performs well on the training data but poorly on the test data. For example, in the image below, the training accuracy is close to 1, while the test accuracy is much lower, less than 0.7. In this case, the model fails to generalize to new samples.
+<div align="center">
+  <img src="../images/overfitting.png" width="400">
+</div> 
+
+### Why overfitting happen
+Overfitting generally happens when the model is too complex relative to the amount of available data. For example, consider a neural network model with a complex structure trained on a small dataset. Some of the reasons that cause overfitting are as follows: 
+
+1- Complex Model: When the model includes many features, it can become overly complex, detecting noise as well.
+
+2- Small test dataset: If the test dataset is small and doesn't adequately represent all possible input data, it may not provide a reliable measure of the model's performance.
+
+3- Training Data Errors and Outliers: Errors and outliers in the training data introduce noise. If the algorithm interprets this noise as meaningful patterns, it can negatively impact the model's ability to generalize to new data.
+
+### Solutions for overfitting
+Several solutions can mitigate  overfitting:
+
+1- Reducing Model Complexity: Simplify the model by decreasing the number of parameters or attributes. For example, in linear regression, include only the most predictive features instead of all available features. As another example, simplifying the structure of neural network models by reducing the number of nodes and layers in a complex architecture.
+
+2- Increasing Sample Size: Gathering more real data can help, although it may be costly. Alternatively, augmentation methods can add samples.
+
+3- Noise Reduction: Clean the data by removing outliers and erroneous data to prevent the model from fitting to noisy data.
+
+4- Regularization: Regularization in linear regression is a technique used to prevent overfitting. When a linear model has large coefficients for its features, it tends to overfit the training data. Regularization terms like L1 and L2 works by penalizing large coefficients, which can decrease them or even set them to zero. This process makes the model smoother and helps prevent overfitting.
```

Comment:
- work instead of works?
- What is L1 and L2? Might be useful to explain, or ignore my comment if it is basic ML :)